x-pack/plugins/enterprise_search/public/applications/shared/react_router_helpers/link_events.test.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { letBrowserHandleEvent } from '../react_router_helpers';
```

Comment:
I imagine if I were TDD'ing `letBrowserHandleEvent`, the unit test you wrote here is the kind of unit test I would write while developing the method. This unit test would help me test the code without having to load everything up in the browser. This would result in faster dev/test loops while building out the method. After the `letBrowserHandleEvent` method is implemented, and these unit tests were passing, I imagine I would then verify it manually in the browser.
👍

Reply:
It can also help you find cases you've missed. I made a small update to `link_events` as a result of writing this test.

Reply:
Just wanted to add here that these are super awesome/thorough tests! 🎉